README.md

```diff
@@ -22,45 +22,221 @@
+You'll need the latest LTS release of nodejs and yarn installed. Assuming that's done, run `yarn` with no args:
 
 ```
-npm install
+yarn
+```
+
+If you'd like to completely clear all packages' `node_modules` and reinstall all deps from scratch, run:
+
+```
+yarn clean-packages
+yarn
+```
+
+### Build the code 🧐
+
+Some code in the repository requires a build step to compile it. To run this build step, use the `qbuild` (quick build) command:
+
+```
+yarn qbuild
+```
+
+The above command does not include dapps because dapps take a long time to build and they have their own scripts to run locally.
+However, if you'd like to build _everything_, you can use the build command:
+
+```
+yarn build
+```
+
+To remove any remnants of previous builds, you can run:
+
+```
+yarn clean
+```
+
+### Run tests 🦾
+
+To run tests, you'll need to start ganache on port 9545:
+
+```
+yarn ganache-cli -e 1000000000 -p 9545 -l 9000000 -d
+```
+
+Note: if you're interested in what these args do:
+
+- `-e` is the amount of ETH to grant the default accounts.
+- `-p` is the port that ganache will listen on.
+- `-d` tells ganache to use a standard set of deterministic accounts on each run.
+
+Then, you can run all of the tests across the repo by running:
+
+```
+yarn test
+```
+
+However, running all of the tests across the repository takes a lot of time. To run the tests for just
+one package, you can run:
+
+```
+yarn workspace <package_name> test
 ```
 
 ### Running the linter 🧽
 
-To run the formatter, run:
+To run the linter in autofix mode (it will attempt to fix any errors it finds), run:
+
+```
+yarn lint-fix
+```
+
+To run the linter in the default mode, where it will print all errors and not modify code, run:
+
+```
+yarn lint
+```
+
+### Packages 📦
+
+Because this repo is a monorepo, it conatains many different npm packages. More will be discussed about these packages in the
+following sections. However, the basic structure is that each pacakge is listed in the `packages/` directory. Each package has
+its own scripts and dependencies and operates (mostly) independently from the others.
+
+### Adding dependencies 👩‍👦
+
+All runtime/production dependencies should be added to the package that needs them. Development dependencies
+should also generally be installed in packages unless they are needed by code that exists outside of any package.
+
+For more details on packages and the monorepo, please see the next section.
+
+To add a dependency to a package:
 
 ```
-npm run lint-fix
+yarn workspace <package_name> add <dependency_name>
 ```
 
-## Coverage 🔎
+Note: development dependencies are those that are not required by the code that's published to the npm registry. If you're not
+sure whether a dependency should be dev or not, just ask! To install a dev dependency in a package:
+
+```
+yarn workspace <package_name> add <dependency_name> --dev
+```
+
+To install a dev dependency at root:
+
+```
+yarn add <dependency_name> --dev
```

Comment:
What's the command to just add a normal dep, similar to `npm i --save X`?

Reply:
Because root doesn’t get published, there should only be dev dependencies at root, so I chose not to include instructions on that. I’ll add a not explaining this.

Reply:
Added a note. PTAL.